### ISSUE TITLE
Adding a Charset on Multipart data causes issues with ASP.net

### ISSUE THREAD

**Issue by mstum** [DotNetOpenAuth/DotNetOpenAuth#262]
I have an application that uploads files to an ASP.net application and it sends in the Content Type including a charset:

```
Content-Type: multipart/form-data; boundary=ea248fb9-996f-4a03-a6a4-09e2a34e94dd; charset=utf-8
```

It seems that ASP.net cannot properly resolve this as Request.Files, and thus Request.Files will be empty.

Intercepting the request and removing the charset:

```
Content-Type: multipart/form-data; boundary=ea248fb9-996f-4a03-a6a4-09e2a34e94dd
```

makes it work.

Is there a way to remove the charset?

This is the OAuth 1.0a stuff:

```
DotNetOpenAuth.OAuth.WebConsumer
```

**Comment by AArnott** [comment]
Are you setting the charset yourself? Or are you saying that WebConsumer is
doing this for you?

Can you try out the
CTPhttps://groups.google.com/forum/?fromgroups#!topic/dotnetopenid/hMvPRtsewsI
that
adds HttpMessageHandlers to OAuth 1 consumers so that you can use
HttpClient's own multi-part POST code rather than the one in DNOA to see if
that solves your problem?

On Wednesday, March 20, 2013, Michael Stum wrote:

> I have an application that uploads files to an ASP.net application and it
> sends in the Content Type including a charset:
> 
> Content-Type: multipart/form-data; boundary=ea248fb9-996f-4a03-a6a4-09e2a34e94dd; charset=utf-8
> 
> It seems that ASP.net cannot properly resolve this as Request.Files, and
> thus Request.Files will be empty.
> 
> Intercepting the request and removing the charset:
> 
> Content-Type: multipart/form-data; boundary=ea248fb9-996f-4a03-a6a4-09e2a34e94dd
> 
> makes it work.
> 
> Is there a way to remove the charset?
> 
> This is the OAuth 1.0a stuff:
> 
> DotNetOpenAuth.OAuth.WebConsumer
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/DotNetOpenAuth/DotNetOpenAuth/issues/262
> .

## 

## 

Andrew Arnott
"I [may] not agree with what you have to say, but I'll defend to the death
your right to say it." - S. G. Tallentyre

**Comment by mstum** [comment]
WebConsumer is doing that automatically. When PrepareAuthorizedRequest is called, it eventually calls MessagingUtilities.PostMultipartNoGetResponse which adds the (hardcoded) Channel.PostEntityEncoding, which is UTF-8.

For now, I've created a HttpModule on my server that removes the charset which works as a workaround, I'll give the 3.4 CTP a try.

**Comment by AArnott** [comment]
This is fixed in v5.0 (which isn't out yet).
